### PR TITLE
Feat/add usernames

### PR DESCRIPTION
**Pull request by asbhaskar**
This requires some database changes to add temp usernames for every user. I also added back the '@' in front of usernames to help differentiate them from names

### CODE REVIEW

File: src/forms/SignUpForm/index.tsx [hunk 1]
```diff
@@ -61,6 +70,18 @@ const SignUpForm: StyledComponent = styled(({ className }) => {
...
+                <Form.Control
+                    id="username"
+                    name="username"
+                    type="text"
+                    placeholder={snippet('username.placeholder')}
+                    onChange={handleChange}
+                    value={username}
+                    required
+                />
+            </Form.Group>
```
Comment (jadeallencook) [inline src/forms/SignUpForm/index.tsx:84]: I'm thinking we should remove this from the signup form. 

They should be able to add a username only from the dashboard.

File: src/forms/EditAccountForm/index.tsx [hunk 2]
```diff
@@ -63,6 +63,18 @@ const EditAccountForm: StyledComponent = styled(
...
+                    <Form.Label>Public Username</Form.Label>
+                    <Form.Control
+                        id="username"
+                        name="username"
+                        type="text"
+                        onChange={handleChange}
+                        value={username}
+                        required
+                        disabled={isSubmitting}
+                    />
```
Comment (jadeallencook) [inline src/forms/EditAccountForm/index.tsx:76]: The only other thing we need to do is check to ensure that no one else has that username.

**Review by jadeallencook — APPROVED** [review]
PTAL @asbhaskar

File: src/forms/SignUpForm/index.tsx (hunk 1 above)
Comment (asbhaskar) [reply src/forms/SignUpForm/index.tsx:84]: This requires us to set a default username for the user's comments (which I don't think should default to real name). It also introduces the possibility that the user will just never see or notice the add username function on the dashboard.

File: src/forms/EditAccountForm/index.tsx (hunk 2 above)
Comment (asbhaskar) [reply src/forms/EditAccountForm/index.tsx:76]: good catch

File: src/forms/SignUpForm/index.tsx (hunk 1 above)
Comment (jadeallencook) [reply src/forms/SignUpForm/index.tsx:84]: Maybe we could pop up a modal that can only be closed after they add a username? 

Or they're not able to leave a comment until they set a username? 

I'm not sure on how to go about this, I'll have to think about it a little more. 

Right now we display their UID if they don't have a name set in their account. 

Also, we need to make sure we verify their username doesn't contain any special characters.

**Pull request closed without merging**